File: LATPRAK4DPBO/C++/Car.cpp
// Saya Marvel Ravindra Dioputra [2200481] LatPrak4 C++ dalam Mata Kuliah DPBO 
// untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

#pragma once
#include "Vehicle.cpp"  // inherit class vehicle sebagai parent car 

class Car : public Vehicle {
private:
    int jumlahKursi;    // atribut jumlah kursi
    int jumlahPintu;    // atribut jumlah pintu

public:
    // konstruktor
    Car() {

    }

    Car(string platNomor, string merk, string tahunProduksi, string warna, int jumlahKursi, int jumlahPintu) : Vehicle(platNomor, merk, tahunProduksi, warna), jumlahKursi(jumlahKursi), jumlahPintu(jumlahPintu) {}

    // setter atribut class car
    void setJumlahKursi(int jumlahKursi) { 
        this->jumlahKursi = jumlahKursi; 
    }

    void setJumlahPintu(int jumlahPintu) { 
        this->jumlahPintu = jumlahPintu; 
    }

    // getter atribut class car
    int getJumlahKursi() { 
        return jumlahKursi; 
    }
    
    int getJumlahPintu() { 
        return jumlahPintu; 
    }

    // destruktor
    ~Car() {

    }
};

File: LATPRAK4DPBO/python/car.py
# Saya Marvel Ravindra Dioputra [2200481] LatPrak4 Python dalam Mata Kuliah DPBO 
# untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

# inheritance dari class vehicle
from vehicle import Vehicle

# class car
class Car(Vehicle):
    # inisialisasi/instansiasi
    def __init__(self, plat_nomor, merk, tahun_produksi, warna, jumlah_kursi, jumlah_pintu):
        # atribut inheritance dari class vehicle
        super().__init__(plat_nomor, merk, tahun_produksi, warna)   
        self.jumlah_kursi = jumlah_kursi    
        self.jumlah_pintu = jumlah_pintu

    # method menampilkan data vehicle dan car
    def display(self):
        super().display()
        print("Jumlah Kursi:", self.jumlah_kursi)
        print("Jumlah Pintu:", self.jumlah_pintu)
